### ISSUE TITLE
Cluster level logging using Elasticsearch should check that logs from pods on all nodes are ingested into Elasticsearch (Failed 5 times in the last 30 runs. Stability: 83 %)

### ISSUE THREAD

**Issue by quinton-hoole** [kubernetes/kubernetes#9486]
Master tracking bug #9312

**Comment by quinton-hoole** [comment]
May be a duplicate of another issue.  TODO.

**Comment by satnam6502** [comment]
Something seems to be going wrong with either where pods are places, or with pod re-starts.
```
Cluster level logging using Elasticsearch 
  should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:39
[BeforeEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:47
STEP: Creating a kubernetes client
>>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gce-flaky/workspace/.kube/config
STEP: Building a namespace api object
INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78
INFO: Get service account default in ns e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78 failed, ignoring for 5s: serviceaccounts "default" not found
INFO: Service account default in ns e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78 found. (5.008347406s)
STEP: Waiting for a default service account to be provisioned in namespace
INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78
INFO: Service account default in ns e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78 found. (2.589049ms)
[It] should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:39
STEP: Checking the Elasticsearch service exists.
STEP: Checking to make sure the Elasticsearch pods are running
STEP: Checking to make sure we are talking to an Elasticsearch service.
STEP: Checking health of Elasticsearch service.
STEP: Waiting for the pods to succeed.
INFO: Waiting up to 5m0s for pod synthlogger-0                                           status to be success or failure
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (7.378789ms elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (5.011808667s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (10.01606834s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (15.020545238s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (20.024841401s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (25.029023232s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (30.033104277s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (35.037174901s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (40.04169416s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (45.046000665s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (50.050293826s elapsed)
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78' status to be 'success or failure'(found phase: "Pending", readiness: false) (55.054486009s elapsed)
STEP: Saw pod success
INFO: Waiting up to 5m0s for pod synthlogger-1                                           status to be success or failure
STEP: Saw pod success
STEP: Checking all the log lines were ingested into Elasticsearch
[AfterEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:48
STEP: Collecting events from namespace "e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78".
INFO: event for synthlogger-0: {kubelet e2e-flaky-minion-s9hb} pulled: Successfully pulled image "gcr.io/google_containers/pause:0.8.0"
INFO: event for synthlogger-0: {kubelet e2e-flaky-minion-s9hb} created: Created with docker id 2e44a476728f93b609df258c37045bf78c1a438ca811252c444653b9893de633
INFO: event for synthlogger-0: {kubelet e2e-flaky-minion-s9hb} started: Started with docker id 2e44a476728f93b609df258c37045bf78c1a438ca811252c444653b9893de633
INFO: event for synthlogger-0: {kubelet e2e-flaky-minion-s9hb} pulled: Successfully pulled image "gcr.io/google_containers/ubuntu:14.04"
INFO: event for synthlogger-0: {kubelet e2e-flaky-minion-s9hb} created: Created with docker id 1dd3afa4bc2c5fab97e589fd104e07bb584807a3172f9fe5c18e5dc2d7e9fb57
INFO: event for synthlogger-0: {kubelet e2e-flaky-minion-s9hb} started: Started with docker id 1dd3afa4bc2c5fab97e589fd104e07bb584807a3172f9fe5c18e5dc2d7e9fb57
INFO: event for synthlogger-0: {kubelet e2e-flaky-minion-s9hb} killing: Killing 2e44a476728f93b609df258c37045bf78c1a438ca811252c444653b9893de633
INFO: event for synthlogger-1: {kubelet e2e-flaky-minion-wg2w} pulled: Successfully pulled image "gcr.io/google_containers/pause:0.8.0"
INFO: event for synthlogger-1: {kubelet e2e-flaky-minion-wg2w} created: Created with docker id 4d12fe3ab103792d375d3894243f862293a079c8d5c0b8daf0b174f359d65eda
INFO: event for synthlogger-1: {kubelet e2e-flaky-minion-wg2w} started: Started with docker id 4d12fe3ab103792d375d3894243f862293a079c8d5c0b8daf0b174f359d65eda
INFO: event for synthlogger-1: {kubelet e2e-flaky-minion-wg2w} pulled: Successfully pulled image "gcr.io/google_containers/ubuntu:14.04"
INFO: event for synthlogger-1: {kubelet e2e-flaky-minion-wg2w} created: Created with docker id 7f3d326cebe80d1b246016f7058dff9eac1afe186c4d7db8cde59fe5ab6ad273
INFO: event for synthlogger-1: {kubelet e2e-flaky-minion-wg2w} started: Started with docker id 7f3d326cebe80d1b246016f7058dff9eac1afe186c4d7db8cde59fe5ab6ad273
INFO: event for synthlogger-1: {kubelet e2e-flaky-minion-wg2w} killing: Killing 4d12fe3ab103792d375d3894243f862293a079c8d5c0b8daf0b174f359d65eda
STEP: Destroying namespace "e2e-tests-es-logging-cf91e60d-942a-4f66-b4e2-f232aa1b9c78" for this suite.

• Failure [96.205 seconds]
Cluster level logging using Elasticsearch
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:40
  should check that logs from pods on all nodes are ingested into Elasticsearch [It]
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:39

  Got incorrect count for node 1 index 0: 3

  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:320
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
```

**Comment by satnam6502** [comment]
Submitted a PR to relax this test although I am not sure if this gets to the heart of the issue since the pods this test creates should never be restarted. I suspect there might be an underlying problem with our system.

**Comment by satnam6502** [comment]
Note that I don't regard this test as 'flaky'. It is detecting some "jitter" in our system which I think is a problem.

**Comment by satnam6502** [comment]
PR #9556 reverted because we should really get to the root of the system instability -- specifically -- jitter.

**Comment by quinton-hoole** [comment]
So we still have what appears to be a case of pods restarting when they should not. Satnam, are you planning to dig into this?  If not, feel free to unassign yourself and I'll try to do it myself, or find someone else.

**Comment by satnam6502** [comment]
I shall look into this since I have this test at hand which I can use to both induce the symptom and help diagnose it.

**Comment by dchen1107** [comment]
@satnam I believe ElasticSearch test failed due to the same reason I found at  #9345. There is a pending pr #9676 to resolve that one. Once that is merged, ElasticSearch test should use the same util method to check which node is ready, and schedule the pod to those ready nodes.

**Comment by quinton-hoole** [comment]
Since merging #9727 (Jun 12, 4:43 PM) this test has failed four times out of ~35 runs (i.e. ~11%) , with the following errors (twice each):

========

/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:39
Failed to find all 200 log lines

=========

/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:39
Expected error:
    <*errors.errorString | 0xc2080bafa0>: {
        s: "failed to unmarshal Elasticsearch response: invalid character 'E' looking for beginning of value",
    }
    failed to unmarshal Elasticsearch response: invalid character 'E' looking for beginning of value
not to have occurred

==============

**Comment by satnam6502** [comment]
OK, it looks like Elasticsearch is giving an error response in an unexpected way. I will investigate.

**Comment by satnam6502** [comment]
After conferring with @thockin the suspicion now is that we might get into a state where one ES replica is ready but the other is not and the service incorrectly maps requests to the not-ready ES replica. I will add diagnostics for this.

**Comment by quinton-hoole** [comment]
Fixed by #10453

**Comment by wojtek-t** [comment]
This test is still flaky - see e.g.
- e2e-gce run 7631
- e2e-gce run 7588

**Comment by satnam6502** [comment]
First of all, I do not like this test being called flaky. It is not flaky: it is actually doing its job a test: it's solid.

Why did the invocation fail? Because three nodes were detected (which means something went wrong before and there was probably bad cleanup). The test puts a synthetic logging source on each node that emits 100 log lines. The test makes sure it detects 100 log lines from each node i.e. 300 expected in this case (usually 200 when the test cluster has not been broken). 
The test detects 200 lines.
Then it detects 100 line i.e. one of the pods or nodes is in trouble.
Then it times out.

Something is wrong. But the test is not wrong or flaky.  Our test environment and procedure is flaky.

```
Cluster level logging using Elasticsearch 
  should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46
[BeforeEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:48
STEP: Creating a kubernetes client
>>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/config
STEP: Building a namespace api object
INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-es-logging-97o7w
INFO: Get service account default in ns e2e-tests-es-logging-97o7w failed, ignoring for 5s: serviceaccounts "default" not found
INFO: Service account default in ns e2e-tests-es-logging-97o7w with secrets found. (5.006238671s)
STEP: Waiting for a default service account to be provisioned in namespace
INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-es-logging-97o7w
INFO: Service account default in ns e2e-tests-es-logging-97o7w with secrets found. (3.043386ms)
[BeforeEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:42
[It] should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46
STEP: Checking the Elasticsearch service exists.
STEP: Checking to make sure the Elasticsearch pods are running
INFO: Waiting up to 5m0s for pod elasticsearch-logging-v1-k31c6                          status to be running
INFO: Found pod 'elasticsearch-logging-v1-k31c6' on node 'e2e-gce-minion-inu5'
INFO: Waiting up to 5m0s for pod elasticsearch-logging-v1-k42t3                          status to be running
INFO: Found pod 'elasticsearch-logging-v1-k42t3' on node 'e2e-gce-minion-iveb'
STEP: Checking to make sure we are talking to an Elasticsearch service.
STEP: Checking health of Elasticsearch service.
INFO: Found 3 nodes.
INFO: Node e2e-gce-minion-inu5 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-16 14:00:40 -0700 PDT
INFO: Successfully found node e2e-gce-minion-inu5 readiness to be true
INFO: Node e2e-gce-minion-iveb condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-16 14:47:32 -0700 PDT
INFO: Successfully found node e2e-gce-minion-iveb readiness to be true
INFO: Node e2e-gce-minion-u7eh condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-16 14:33:08 -0700 PDT
INFO: Successfully found node e2e-gce-minion-u7eh readiness to be true
INFO: Found 3 healthy nodes.
STEP: Waiting for the pods to succeed.
INFO: Waiting up to 5m0s for pod synthlogger-0                                           status to be success or failure
INFO: No Status.Info for container 'synth-logger' in pod 'synthlogger-0' yet
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-97o7w' status to be 'success or failure'(found phase: "Pending", readiness: false) (189.249356ms elapsed)
INFO: Nil State.Terminated for container 'synth-logger' in pod 'synthlogger-0' in namespace 'e2e-tests-es-logging-97o7w' so far
INFO: Waiting for pod synthlogger-0                                           in namespace 'e2e-tests-es-logging-97o7w' status to be 'success or failure'(found phase: "Running", readiness: false) (5.19375049s elapsed)
STEP: Saw pod success
INFO: Waiting up to 5m0s for pod synthlogger-1                                           status to be success or failure
STEP: Saw pod success
INFO: Waiting up to 5m0s for pod synthlogger-2                                           status to be success or failure
INFO: Nil State.Terminated for container 'synth-logger' in pod 'synthlogger-2' in namespace 'e2e-tests-es-logging-97o7w' so far
INFO: Waiting for pod synthlogger-2                                           in namespace 'e2e-tests-es-logging-97o7w' status to be 'success or failure'(found phase: "Running", readiness: false) (3.711349ms elapsed)
STEP: Saw pod success
STEP: Checking all the log lines were ingested into Elasticsearch
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 76.145576ms expecting to find 300 log lines but saw only 0
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 10.190427609s expecting to find 300 log lines but saw only 100
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 20.226817955s expecting to find 300 log lines but saw only 100
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 30.301266765s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 53.67460275s expecting to find 300 log lines but saw only 100
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 1m3.714941966s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 1m13.755686936s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 1m23.794308535s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 1m33.83001187s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 1m43.877238235s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 1m53.932492406s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 2m3.962292659s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 2m13.992785067s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 2m24.025234148s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 2m34.055092649s expecting to find 300 log lines but saw only 200
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 2m44.087797631s expecting to find 300 log lines but saw only 100
INFO: pod 0: elasticsearch-logging-v1-k31c6 PodIP 10.245.0.4 phase Running condition [{Type:Ready Status:True}]
INFO: pod 1: elasticsearch-logging-v1-k42t3 PodIP 10.245.3.4 phase Running condition [{Type:Ready Status:True}]
INFO: After 2m54.12289849s expecting to find 300 log lines but saw only 100
[AfterEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:49
STEP: Collecting events from namespace "e2e-tests-es-logging-97o7w".
INFO: event for synthlogger-0: {kubelet e2e-gce-minion-inu5} pulled: Pod container image "gcr.io/google_containers/pause:0.8.0" already present on machine
INFO: event for synthlogger-0: {kubelet e2e-gce-minion-inu5} created: Created with docker id 9f93b56a6b80
INFO: event for synthlogger-0: {kubelet e2e-gce-minion-inu5} started: Started with docker id 9f93b56a6b80
INFO: event for synthlogger-0: {kubelet e2e-gce-minion-inu5} pulled: Successfully pulled image "gcr.io/google_containers/ubuntu:14.04"
INFO: event for synthlogger-0: {kubelet e2e-gce-minion-inu5} created: Created with docker id 13c834a7027e
INFO: event for synthlogger-0: {kubelet e2e-gce-minion-inu5} started: Started with docker id 13c834a7027e
INFO: event for synthlogger-0: {kubelet e2e-gce-minion-inu5} killing: Killing with docker id 9f93b56a6b80
INFO: event for synthlogger-1: {kubelet e2e-gce-minion-iveb} pulled: Pod container image "gcr.io/google_containers/pause:0.8.0" already present on machine
INFO: event for synthlogger-1: {kubelet e2e-gce-minion-iveb} created: Created with docker id e70a51920483
INFO: event for synthlogger-1: {kubelet e2e-gce-minion-iveb} started: Started with docker id e70a51920483
INFO: event for synthlogger-1: {kubelet e2e-gce-minion-iveb} pulled: Successfully pulled image "gcr.io/google_containers/ubuntu:14.04"
INFO: event for synthlogger-1: {kubelet e2e-gce-minion-iveb} created: Created with docker id d18c95877b75
INFO: event for synthlogger-1: {kubelet e2e-gce-minion-iveb} started: Started with docker id d18c95877b75
INFO: event for synthlogger-1: {kubelet e2e-gce-minion-iveb} killing: Killing with docker id e70a51920483
INFO: event for synthlogger-2: {kubelet e2e-gce-minion-u7eh} pulled: Pod container image "gcr.io/google_containers/pause:0.8.0" already present on machine
INFO: event for synthlogger-2: {kubelet e2e-gce-minion-u7eh} created: Created with docker id 2eb6643291a0
INFO: event for synthlogger-2: {kubelet e2e-gce-minion-u7eh} started: Started with docker id 2eb6643291a0
INFO: event for synthlogger-2: {kubelet e2e-gce-minion-u7eh} pulled: Successfully pulled image "gcr.io/google_containers/ubuntu:14.04"
INFO: event for synthlogger-2: {kubelet e2e-gce-minion-u7eh} created: Created with docker id 0ce151f3aad3
INFO: event for synthlogger-2: {kubelet e2e-gce-minion-u7eh} started: Started with docker id 0ce151f3aad3
INFO: event for synthlogger-2: {kubelet e2e-gce-minion-u7eh} killing: Killing with docker id 2eb6643291a0
INFO: Waiting up to 1m0s for all nodes to be ready
INFO: Node e2e-gce-minion-inu5 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-16 14:00:40 -0700 PDT
INFO: Successfully found node e2e-gce-minion-inu5 readiness to be true
INFO: Node e2e-gce-minion-iveb condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-16 14:47:32 -0700 PDT
INFO: Successfully found node e2e-gce-minion-iveb readiness to be true
INFO: Node e2e-gce-minion-u7eh condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-07-16 14:33:08 -0700 PDT
INFO: Successfully found node e2e-gce-minion-u7eh readiness to be true
STEP: Destroying namespace "e2e-tests-es-logging-97o7w" for this suite.

• Failure [245.040 seconds]
Cluster level logging using Elasticsearch
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:47
  should check that logs from pods on all nodes are ingested into Elasticsearch [It]
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46

  Failed to find all 300 log lines

  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:372
```

**Comment by satnam6502** [comment]
I need to run now, but I will try to dig into this deeper later.

**Comment by mbforbes** [comment]
Maybe a silly question, but is this fixed? Do we still have problems with the test environment and procedure? Do we need more help here?

**Comment by satnam6502** [comment]
Looking at Jenkins I see:
```
Cluster level logging using Elasticsearch should check that logs from pods on all nodes are ingested into Elasticsearch (No known failures. Stability 100 %) No flake or failure	1 min 11 sec	Passed
```

**Comment by satnam6502** [comment]
I think the last failure occurred on 1 August and I blame something other than the test for that :-)

```
Cluster level logging using Elasticsearch 
  should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46
[BeforeEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:50
STEP: Creating a kubernetes client
>>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/config
STEP: Building a namespace api object
INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-es-logging-o1t3c
INFO: Get service account default in ns e2e-tests-es-logging-o1t3c failed, ignoring for 5s: serviceaccounts "default" not found
INFO: Service account default in ns e2e-tests-es-logging-o1t3c with secrets found. (5.005956881s)
STEP: Waiting for a default service account to be provisioned in namespace
INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-es-logging-o1t3c
INFO: Service account default in ns e2e-tests-es-logging-o1t3c with secrets found. (2.032036ms)
[BeforeEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:42
[It] should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46
STEP: Checking the Elasticsearch service exists.
STEP: Checking to make sure the Elasticsearch pods are running
INFO: Waiting up to 5m0s for pod elasticsearch-logging-v1-6hcii                          status to be running
INFO: Found pod 'elasticsearch-logging-v1-6hcii' on node 'e2e-gce-minion-4qyp'
INFO: Waiting up to 5m0s for pod elasticsearch-logging-v1-k31vl                          status to be running
INFO: Found pod 'elasticsearch-logging-v1-k31vl' on node 'e2e-gce-minion-jzby'
STEP: Checking to make sure we are talking to an Elasticsearch service.
STEP: Checking health of Elasticsearch service.
INFO: Bad JSON: Error: 'dial tcp 10.245.0.3:9200: connection refused'
Trying to reach: 'http://10.245.0.3:9200/_cluster/health?health=pretty'
[AfterEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/framework.go:51
STEP: Collecting events from namespace "e2e-tests-es-logging-o1t3c".
INFO: Waiting up to 1m0s for all nodes to be ready
INFO: Node e2e-gce-minion-4qyp condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-08-01 14:19:42 -0700 PDT
INFO: Successfully found node e2e-gce-minion-4qyp readiness to be true
INFO: Node e2e-gce-minion-jzby condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-08-01 14:19:42 -0700 PDT
INFO: Successfully found node e2e-gce-minion-jzby readiness to be true
INFO: Node e2e-gce-minion-tig5 condition 1/1: type: Ready, status: True, reason: "kubelet is posting ready status", message: "", last transistion time: 2015-08-01 14:19:42 -0700 PDT
INFO: Successfully found node e2e-gce-minion-tig5 readiness to be true
STEP: Destroying namespace "e2e-tests-es-logging-o1t3c" for this suite.

• Failure [15.331 seconds]
Cluster level logging using Elasticsearch
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:47
  should check that logs from pods on all nodes are ingested into Elasticsearch [It]
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46

  Expected error:
      <*errors.errorString | 0xc2081bf830>: {
          s: "failed to unmarshal Elasticsearch response: invalid character 'E' looking for beginning of value",
      }
      failed to unmarshal Elasticsearch response: invalid character 'E' looking for beginning of value
  not to have occurred

  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:163
```

**Comment by quinton-hoole** [comment]
Test is 100% stable now.  Closing.

**Comment by satnam6502** [comment]
Great news! Thanks.
